OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -64,8 +67,15 @@ public IEnumerable<string> ProvidesPrerequisites
 
 		public void OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
 		{
-			if (info.ResetOnOwnerChange)
-				Update(newOwner, newOwner.Faction.InternalName);
+			if (!info.ResetOnOwnerChange)
+				return;
+
+			// The TechTree of the oldOwner gets updated in the Update method
+			Update(newOwner, newOwner.Faction.InternalName);
+
+			// Now refresh the TechTree of the new owner
+			techTree = newOwner.PlayerActor.Trait<TechTree>();
+			techTree.Update();
```

Comment:
These shouldnt be needed - the tech tree code should already be detecting the prereq change and updating itself. Find what is broken there and dlfix that instead.